ims-apple/IMS Apple/Views/AddingProduct/AddingProductView.swift

```diff
@@ -0,0 +1,172 @@
+    static let horizontalSpacing: CGFloat = 20
+    static let hasBorder: Bool = true
+    static let minHeight: CGFloat = 30
+    static let cornerRadiusSize: CGFloat = 10
+    static let photoMaxHeight: CGFloat = 120
+    static let detailButtonHeight: CGFloat = 70
+    static let strokLengths: CGFloat = 10
+    static let spaceSize: CGFloat = 10
+    static let loadingImage: String = "square.and.arrow.down"
+    static let saveProductImage: String = "square.and.arrow.down.on.square.fill"
+}
+
+struct AddingProductView: View {
+    @State private var productName: String = ""
+    @State private var price: String = ""
+    @State private var availableUnits: String = ""
+    @State private var productDetails: String = ""
+    @State private var state: String = ""
+    @State private var selectedState: String = "Disponible"
+    @State private var showingImagePicker = false
+    @State private var avatarItem: PhotosPickerItem?
+    @State private var avatarImage: Image?
+    
+    var body: some View {
+        VStack {
+            HStack {
+                Text("Datos generales")
+                    .font(.headline)
+                
+                Spacer()
+                
+                Button("Guardar Product") { }
+                    .buttonStyle(GradientButtonStyle(imageLeft: Constants.saveProductImage,
+                                                     buttonHeight: Constants.buttonHeight))
+            }
+            .frame(maxWidth: .infinity, alignment: .leading)
+            
+            HStack(spacing: Constants.horizontalSpacing) {
+                productInformationView
+                
+                productDetailsView
+            }
+            
+            Spacer()
+        }
+        .padding()
+        .frame(maxWidth: .infinity)
+        .background(.grayBackground)
+    }
+    
+    // MARK: - Product Information
+    
+    private var productInformationView: some View {
+        VStack {
+            IMSTextField(
+                type: .custom("Nombre del producto"),
+                text: $productName,
+                hasBorder: Constants.hasBorder,
+                maxWidth: .infinity
+            )
+            
+            IMSTextField(
+                type: .custom("Precio"),
+                text: $price,
+                hasBorder: Constants.hasBorder,
+                maxWidth: .infinity
+            )
+            
+            IMSTextField(
+                type: .custom("Unidades disponibles"),
+                text: $availableUnits,
+                hasBorder: Constants.hasBorder,
+                maxWidth: .infinity
+            )
+            
+            Text("Estado")
+                .frame(maxWidth: .infinity, alignment: .leading)
+            
+            Menu {
+                ForEach(["Disponible", "Ocupado", "Agotado"], id: \.self) { item in
+                    Button(item) {
+                        withAnimation {
+                            selectedState = item
```

Comment:
NIT: We can rename the `item` variable to `productState`.

```suggestion
                ForEach(["Disponible", "Ocupado", "Agotado"], id: \.self) { productState in
                    Button(productState) {
                        withAnimation {
                            selectedState = productState
```